core/local/atom_watcher.js

```diff
@@ -106,13 +109,16 @@ module.exports = class AtomWatcher {
     dispatch.loop(buffer, stepOptions)
   }
 
-  start () {
+  async start () {
     log.debug('starting...')
     this.running = new Promise((resolve, reject) => {
       this._runningResolve = resolve
       this._runningReject = reject
     })
     this.events.emit('local-start')
+    await Promise.all(steps.map(async step => {
+      if (step.initialState) _.assign(this.state, await step.initialState(this))
+    }))
```

Comment:
Maybe extract function? E.g.:

```js
const stepsInitialState = opts =>
  Promise.reduce(
    async (state, step) =>
      step.initialState
        ? Object.assign(state, await step.initialState(opts))
        : state
    steps,
    {}
  )
```

Then in `AtomWatcher` constructor:

```js
this.state = await stepsInitialState(this)
```

Reply:
The step initialisation needs to happen in `start()` otherwise we won't reset the state after restarting the watcher. That's the whole point of this PR ;-).

Reply:
`s/constructor/start` (my bad :stuck_out_tongue:)